mesheryctl/internal/cli/root/credentials/create.go

```diff
@@ -0,0 +1,157 @@
+// Copyright Meshery Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package credentials
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshery/server/models"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/manifoldco/promptui"
+)
+
+var createCredentialCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Create a new credential",
+	Long:  `Create a new credential by providing the name, user ID, type, and secret of the credential`,
+	Example: `
+// Create a new credential
+mesheryctl exp credential create
+`,
+	Args: cobra.MinimumNArgs(0),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
```

Comment:
Could you please consider moving this logic into the PreRunE section?

Reply:
Done